src/hooks/formHooks/program/useNewProgramForm.tsx

```diff
@@ -113,12 +91,27 @@ const programSchema = yup.object().shape({
   [AddProgramFormFields.PROGRAM_PRICE]: yup.number().required().min(0),
   [AddProgramFormFields.PROGRAM_DESCRIPTION]: yup.string().min(20).max(150),
 });
+type UseProgramFormProps = {
+  editProgram?: Program;
+};
 
-export const useNewProgramForm = () => {
+export const useNewProgramForm = ({ editProgram }: UseProgramFormProps) => {
   const [pending, setPending] = useState(false);
   const { mutateAsync: addQueryProgram } = useAddProgramMutation();
+  const { mutateAsync: updateQueryProgram } = useUpdateProgramMutation("s");
   const { user } = useUserContext();
 
+  const defaultValues = editProgram
+    ? {
+        ...defaultProgramValues,
```

Comment:
why defaultValues?
Is it possible that during editing any value will be undefined?
we can go through all the values and assign one to the appropriate keys instead of doing it manually value by value